Source/JavaScriptCore/runtime/JSONObject.cpp

```diff
@@ -627,6 +637,517 @@ bool Stringifier::Holder::appendNextProperty(Stringifier& stringifier, StringBui
+                auto* characters = name.characters8();
+                for (unsigned i = 0; i < nameLength; ++i) {
+                    auto character = characters[i];
+                    if (WTF::escapedFormsForJSON[character]) {
+                        recordFailure("property name character needs escaping"_s);
+                        return false;
+                    }
+                    m_buffer[m_length + 1 + i] = character;
+                }
+                m_buffer[m_length + 1 + nameLength] = '"';
+                m_buffer[m_length + 1 + nameLength + 1] = ':';
+                m_length += 1 + nameLength + 2;
+                append(object.getDirect(entry.offset()));
+                return !haveFailure();
+            });
+            if (haveFailure())
+                return;
+        } else {
+#if !FAST_STRINGIFY_GETTER_EXPERIMENT
+            recordFastPropertyEnumerationFailure(object);
+            return;
+#else
+            auto scope = DECLARE_THROW_SCOPE(m_vm);
+            PropertyNameArray names(m_vm, PropertyNameMode::Strings, PrivateSymbolMode::Exclude);
+            object.methodTable()->getOwnPropertyNames(&object, &m_globalObject, names, DontEnumPropertiesMode::Exclude);
+            if (UNLIKELY(scope.exception())) {
+                scope.clearException();
+                recordFailure("getOwnPropertyNames exception"_s);
+                return;
+            }
+            for (auto& name : names) {
+                if (!name.string().is8Bit()) {
+                    recordFailure("16-bit property name"_s);
+                    return;
+                }
+                bool needComma = m_buffer[m_length - 1] != '{';
+                unsigned nameLength = name.length();
+                if (remainingCapacity() < needComma + 1 + nameLength + 2) {
+                    recordBufferFull();
+                    return;
+                }
+                if (needComma)
+                    m_buffer[m_length++] = ',';
+                m_buffer[m_length] = '"';
+                auto* characters = name.string().characters8();
+                for (unsigned i = 0; i < nameLength; ++i) {
+                    auto character = characters[i];
+                    if (mayNeedJSONEscaping(character)) {
+                        recordFailure("property name character needs escaping"_s);
+                        return;
+                    }
+                    m_buffer[m_length + 1 + i] = character;
+                }
+                m_buffer[m_length + 1 + nameLength] = '"';
+                m_buffer[m_length + 1 + nameLength + 1] = ':';
+                m_length += 1 + nameLength + 2;
+                scope.assertNoException();
+                auto value = object.get(&m_globalObject, name);
+                if (UNLIKELY(scope.exception())) {
+                    scope.clearException();
+                    recordFailure("get exception"_s);
+                    return;
+                }
```

Comment:
Do we have a side-effect on this? For example, if there is a getter, calling this can be user-observable. So, if

1. First use FastJSONStringfier
2. Fail
3. Use generic Stringifier

Then, user can observe (1) and (2) since getter is called twice.

Reply:
Yes, this was part of the deleted experiment. That’s why it was disabled.